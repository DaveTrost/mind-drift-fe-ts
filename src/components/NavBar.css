.NavBar {
  margin-top: 2em;
  button {
    border: solid var(--light-blue) 2px;
    border-radius: 8px;
    width: 8em;
    height: 2.3em;
    background-color: white;
    font-family: Montserrat;
    font-weight: 200;
    font-size: 12px;
    margin-left: 1.5em;
  }

  button:hover {
    background-color: var(--light-blue);
    border: none;
    color: white;
    cursor: pointer;
  }
}

.NavigationFooter {
  display: flex;
  position: fixed;
  bottom: 0;
  background-color: var(--light-grey);
  width: 100vw;
  left: 0;
  a {
    text-decoration: none;
    color: white;
    font-family: Montserrat;
    margin: 1em;
    font-weight: 200;
    font-size: 14px;
  }
  a:hover {
    font-weight: 400;
  }
}