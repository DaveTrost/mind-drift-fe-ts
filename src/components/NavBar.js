import React from 'react';
import { useAuth0 } from '../react-auth0-spa';
import { Link } from 'react-router-dom';
import styles from './NavBar.css';

const NavBar = () => {
  const { isAuthenticated, loginWithRedirect, logout } = useAuth0();

  return (
    <div className={styles.NavBar}>
      {!isAuthenticated && (
        <button onClick={() => loginWithRedirect({})}>Log In</button>
      )}
      {isAuthenticated && <button onClick={() => logout()}>Log Out</button>}
      {isAuthenticated && (
        <span className={styles.NavigationFooter}>
          <Link to="/"><img src="/src/assets/HOMEicon.png" alt="Home" /></Link>&nbsp;
          <Link to="/settings"><img src="/src/assets/SETTINGSicon.png" alt="Settings" /></Link>
          <Link to="/profile"><img src="/src/assets/PROFILEicon.png" alt="Profile" /></Link>
        </span>
      )}  
    </div>
  );
};

export default NavBar;
