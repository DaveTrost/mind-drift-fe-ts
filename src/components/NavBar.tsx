import * as React from 'react';
import { Link } from 'react-router-dom';

const { useLogout } = require('../utils/WithSession');
const styles = require('./NavBar.css');
const HOMEicon = require('../assets/HOMEicon.png');
const PROFILEicon = require('../assets/PROFILEicon.png');
const SETTINGSicon = require('../assets/SETTINGSicon.png');
const FLOWERicon = require('../assets/FLOWERicon.png');

const NavBar = () => {
  const logout = useLogout();

  return (
    <div className={styles.NavBar}>
      <button onClick={logout}>Log Out</button>
      <span className={styles.NavigationFooter}>
        <Link to="/"><img src={HOMEicon} alt="Home" /></Link>&nbsp;
        <Link to="/profile"><img src={PROFILEicon} alt="Profile" /></Link>
        <Link to="/settings"><img src={SETTINGSicon} alt="Settings" /></Link>
        <Link to="/about-us"><img src={FLOWERicon} alt="About Us" /></Link>
      </span>
    </div>
  );
};

export default NavBar;
