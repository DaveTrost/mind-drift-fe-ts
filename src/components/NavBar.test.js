import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';
import useAuth0Mock from '../react-auth0-spa';

jest.mock('../react-auth0-spa', () => ({}));

describe('Nav Bar component', () => {
  it('renders Nav Bar', () => {
    useAuth0Mock.useAuth0 = () => ({
      isAuthenticated: true,
      loginWithRedirect: () => true,
      logout: () => false,
    });
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
