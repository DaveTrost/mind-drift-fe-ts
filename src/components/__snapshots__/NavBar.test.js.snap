// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Bar component renders Nav Bar 1`] = `
<div
  className="NavBar"
>
  <button
    onClick={[Function]}
  >
    Log Out
  </button>
  <span
    className="NavigationFooter"
  >
    <Link
      to="/"
    >
      <img
        alt="Home"
        src="https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/HOMEicon.png"
      />
    </Link>
    Â 
    <Link
      to="/settings"
    >
      <img
        alt="Settings"
        src="https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/SETTINGSicon.png"
      />
    </Link>
    <Link
      to="/profile"
    >
      <img
        alt="Profile"
        src="https://raw.githubusercontent.com/mindDrift/mind-drift-fe/dev/src/assets/PROFILEicon.png"
      />
    </Link>
  </span>
</div>
`;
